src/providers/token-provider.ts

```diff
@@ -660,15 +660,21 @@ export class TokenProvider implements ITokenProvider {
         }
       ]);
 
-      result = this.multicall2Provider.callSameFunctionOnMultipleContracts<
-        undefined,
-        [string]
-      >({
-        addresses,
-        contractInterface: bytes32Interface,
-        functionName: 'symbol',
-        providerConfig,
-      });
+      try {
+        result = this.multicall2Provider.callSameFunctionOnMultipleContracts<
+          undefined,
+          [string]
+        >({
+          addresses,
+          contractInterface: bytes32Interface,
+          functionName: 'symbol',
+          providerConfig,
+        });
+      } catch (error) {
+        log.fatal({ addresses }, `TokenProvider.getTokenSymbol[bytes32] failed with error ${error}.`);
+
+        return Promise.reject(new Error('[TokenProvider.getTokenSymbol] Impossible to fetch token symbol.'));
```

Comment:
You can just throw new Error - its the same as rejecting the promise if you throw in an async function